File: Java/3202. Find the Maximum Length of Valid Subsequence II.java
class Solution {
    public int maximumLength(int[] nums, int k) {
        int ans = 0;
        for (int c2 = 0; c2 < k; c2++) {
            int[] dp = new int[k];
            for (int c1: nums) {
                c1 %= k;
                dp[c1] = dp[(c2 - c1 + k) % k] + 1;
                ans = Math.max(ans, dp[c1]);
            }
        }
        return ans;
    }
}

File: Python3/3202. Find the Maximum Length of Valid Subsequence II.py
class Solution:
    def maximumLength(self, nums: List[int], k: int) -> int:
        ans = 0

        for c2 in range(k):
            dp = [0] * k
            for c1 in nums:
                c1 %= k
                dp[c1] = dp[(c2 - c1) % k] + 1
                ans = max(ans, dp[c1])

        return ans